examples/async.rs

```diff
@@ -1,16 +1,25 @@
 extern crate futures;
 extern crate reqwest;
 extern crate tokio_core;
+extern crate tokio_io;
 
+use std::io;
 use futures::Future;
 
 fn main() {
     let mut core = tokio_core::reactor::Core::new().unwrap();
     let client = reqwest::unstable::async::Client::new(&core.handle()).unwrap();
 
-    let work = client.get("https://hyper.rs").unwrap().send().map(|res| {
-        println!("{}", res.status());
-    });
+    let work = client.get("http://localhost:8080/range/4096?duration=5&chunk_size=300")
```

Comment:
Thanks, I'll work on the server and revert these examples :+1:

> What does duration do? Is that a timeout between chunks?

Yeh this is currently just pointing at a local httpbin (the hosted one is behind nginx so it won't actually chunk responses). Yeh from memory it's a delay between the chunks. It'd be nice to be able to tell the server when to send the next chunk so we can force `WouldBlock` imperatively, but don't want to complicate it too much.